node/pod.go

```diff
@@ -213,8 +213,11 @@ func (pc *PodController) enqueuePodStatusUpdate(ctx context.Context, q workqueue
 		if obj, ok := pc.knownPods.Load(key); ok {
 			kpod := obj.(*knownPod)
 			kpod.Lock()
+			defer kpod.Unlock()
+			if cmp.Equal(kpod.lastPodStatusReceivedFromProvider, pod) {
+				return
+			}
 			kpod.lastPodStatusReceivedFromProvider = pod
-			kpod.Unlock()
 			q.AddRateLimited(key)
```

Comment:
Dumb question, do you know if this function blocks?

Reply:
I check the implementation. This `q` is base on a `chan` with size `1000`. So if not full, would not block.